### PR TITLE
Fix/cppcheck errors

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Related task: [APPLINK-31124](https://adc.luxoft.com/jira/browse/APPLINK-31124)

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@LuxoftAKutsan please review

File: src/components/transport_manager/src/transport_manager_impl.cc [hunk 1]
```diff
@@ -618,22 +618,30 @@ void TransportManagerImpl::AddConnection(const ConnectionInternal& c) {
...
     const uint32_t id, transport_adapter::TransportAdapter* transport_adapter) {
   LOG4CXX_AUTO_TRACE(logger_);
   LOG4CXX_DEBUG(logger_, "Id: " << id);
   sync_primitives::AutoWriteLock lock(connections_lock_);
-  for (std::vector<ConnectionInternal>::iterator it = connections_.begin();
-       it != connections_.end();
-       ++it) {
-    if (it->id == id) {
-      connections_.erase(it);
+  std::vector<ConnectionInternal>::iterator it = std::find_if(connections_.begin(),
```
Comment (LuxoftAKutsan) [inline src/components/transport_manager/src/transport_manager_impl.cc]: @LitvinenkoIra it can be const.

Comment (istoilovgithub) [inline src/components/transport_manager/src/transport_manager_impl.cc]: Seems to be correct according to c++98 specification - vector::erase requires non const.

**Comment by LitvinenkoIra** [comment]
Rebuild required

File: src/components/transport_manager/src/transport_manager_impl.cc (hunk 1 above)
Comment (LuxoftAKutsan) [inline src/components/transport_manager/src/transport_manager_impl.cc]: @istoilovgithub please don't forget to add person who you are adderss you comment. 
I mean 
```
const std::vector<ConnectionInternal>::iterator it
```

Comment (LitvinenkoIra) [inline src/components/transport_manager/src/transport_manager_impl.cc]: Done be43458dcb2adbe0f385a697655cb54c95eac67c

**Comment by LitvinenkoIra** [comment]
Rebuild required

**Comment by LitvinenkoIra** [comment]
Rebuild required

**Pull request merged**